go/vt/vttablet/tabletserver/schema/load_table.go

```diff
@@ -117,20 +102,49 @@ func loadMessageInfo(ta *Table, comment string) error {
 
 	ta.MessageInfo.MaxBackoff, _ = getDuration(keyvals, "vt_max_backoff")
 
+	// configure the columns that will be loaded into memory and streamed to subscribers
+	hiddenCols := map[string]struct{}{
+		"priority":   {},
+		"time_next":  {},
+		"epoch":      {},
+		"time_acked": {},
+	}
+
+	requiredCols := []string{
+		"id",
+		"priority",
+		"time_next",
+		"epoch",
+		"time_acked",
+	}
+
+	// make sure required columns are present
 	for _, col := range requiredCols {
 		num := ta.FindColumn(sqlparser.NewColIdent(col))
 		if num == -1 {
 			return fmt.Errorf("%s missing from message table: %s", col, ta.Name.String())
 		}
 	}
 
-	// Load user-defined columns. Any "unrecognized" column is user-defined.
-	for _, field := range ta.Fields {
-		if _, ok := hiddenCols[strings.ToLower(field.Name)]; ok {
-			continue
+	allowedStreamingCols, _ := getCols(keyvals, "vt_stream_cols")
+
+	// make sure that all the specified columns are present
+	for _, col := range allowedStreamingCols {
+		num := ta.FindColumn(sqlparser.NewColIdent(col))
+		if num == -1 {
+			return fmt.Errorf("%s missing from message table: %s", col, ta.Name.String())
 		}
-		ta.MessageInfo.Fields = append(ta.MessageInfo.Fields, field)
 	}
+
+	if len(allowedStreamingCols) > 0 {
+		if allowedStreamingCols[0] != "id" {
```

Comment:
But I guess we don't need the other requiredCols here? 
```
		"priority",
 		"time_next",
 		"epoch",
 		"time_acked",
```

I guess those are required in the table structure but not the messages... that would make sense. This could be a reason that `vt_message_fields` is the best name...